File: internal/content/testdata/bench/java/merge-intervals.java
import java.util.*;

class Solution {
    public long[][] merge_intervals(long[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Long.compare(a[0], b[0]));
        List<long[]> res = new ArrayList<>();
        for (long[] iv : intervals) {
            if (!res.isEmpty() && iv[0] <= res.get(res.size() - 1)[1]) {
                res.get(res.size() - 1)[1] = Math.max(res.get(res.size() - 1)[1], iv[1]);
            } else {
                res.add(new long[]{iv[0], iv[1]});
            }
        }
        return res.toArray(new long[0][]);
    }
}

File: internal/content/testdata/bench/csharp/merge-intervals.cs
public class Solution {
    public long[][] merge_intervals(long[][] intervals) {
        var sorted = intervals.OrderBy(x => x[0]).ToArray();
        var res = new List<long[]>();
        foreach (var interval in sorted) {
            long s = interval[0], e = interval[1];
            if (res.Count > 0 && s <= res[res.Count - 1][1]) {
                res[res.Count - 1][1] = Math.Max(res[res.Count - 1][1], e);
            } else {
                res.Add(new long[] { s, e });
            }
        }
        return res.ToArray();
    }
}
